internal/auditd/auditd.go

```diff
@@ -102,17 +99,52 @@ func (o *Auditd) Read(ctx context.Context) error {
 		case err = <-parseAuditLogsDone:
 			return fmt.Errorf("audit log parser exited unexpectedly with error - %w", err)
 		case result := <-reassembleAuditdEvents:
-			if result.err != nil {
-				return fmt.Errorf("failed to reassemble auditd event - %w", result.err)
+			if err := consumeReassembledEvent(result, tracker); err != nil {
+				return err
 			}
+		}
+	}
+}
 
-			err = tracker.auditdEvent(result.event)
-			if err != nil {
-				return fmt.Errorf("failed to handle auditd event '%s' seq '%d' - %w",
-					result.event.Type, result.event.Sequence, err)
+func consumeReassembledEvent(result reassembleAuditdEventResult, tracker *sessionTracker) error {
+	if result.err != nil {
+		return fmt.Errorf("failed to reassemble auditd event - %w", result.err)
+	}
+
+	if err := tracker.auditdEvent(result.event); err != nil {
+		return fmt.Errorf("failed to handle auditd event '%s' seq '%d' - %w",
+			result.event.Type, result.event.Sequence, err)
+	}
+
+	return nil
+}
+
+func closeReassembler(
```

Comment:
Can you add a Go doc comment that explains why we need to have a separate function for this? (something similar to the original TODO I wrote would be helpful)

Reply:
Sure.